### PR TITLE
Conseil-89 Define actor systems on app entrypoint and do clean-up

### LINKED ISSUE #89: Rationalize handling of Akka actor systems in codebase
There are two actor systems currently in the codebase. This should be reduced to one if possible. In addition, all actor systems and materializers must also be properly shut down.

### PR DESCRIPTION
**Pull request by ivanopagano**
fix #89 

The project still starts up 2 separate actor systems, because we have different entrypoint to launch: 1. conseil 2. lorre

Now the actor system is created there to avoid possible errors in interactions between dependency with the node interface, which now takes the system as a construction parameter.

Provided cleanup during shutdown for Lorre too.

### CODE REVIEW

**Comment by vishakh** [comment]
We are ready to merge this PR once you resolve the merge conflicts.

File: src/main/scala/tech/cryptonomic/conseil/tezos/TezosNodeInterface.scala [hunk 1]
```diff
@@ -36,14 +36,13 @@ trait TezosRPCInterface {
...
   */
-object TezosNodeInterface extends TezosRPCInterface with LazyLogging {
+case class TezosNodeInterface()(implicit system: ActorSystem) extends TezosRPCInterface with LazyLogging {
 
   private val conf = ConfigFactory.load
   private val awaitTime = conf.getInt("dbAwaitTimeInSeconds").seconds
   private val entityGetTimeout = conf.getInt("GET-ResponseEntityTimeoutInSeconds").seconds
   private val entityPostTimeout = conf.getInt("POST-ResponseEntityTimeoutInSeconds").seconds
 
-  implicit val system: ActorSystem = ActorSystem("lorre-system")
```
[the remark this answers is not in the document]
Comment (ivanopagano) [reply src/main/scala/tech/cryptonomic/conseil/tezos/TezosNodeInterface.scala]: It's been moved into the main application entry-point, in this case the `Lorre.main` method

File: src/main/scala/tech/cryptonomic/conseil/tezos/TezosNodeInterface.scala [hunk 2]
```diff
@@ -36,14 +36,13 @@ trait TezosRPCInterface {
 /**
   * Concrete implementation of the above.
   */
-object TezosNodeInterface extends TezosRPCInterface with LazyLogging {
+case class TezosNodeInterface()(implicit system: ActorSystem) extends TezosRPCInterface with LazyLogging {
```
[the remark this answers is not in the document]
Comment (ivanopagano) [reply src/main/scala/tech/cryptonomic/conseil/tezos/TezosNodeInterface.scala]: Exactly, we provide it, implicitly, from the external class that creates the `TezosNodeInterface` instance, as an external dependency

File: src/main/scala/tech/cryptonomic/conseil/Lorre.scala [hunk 3]
```diff
@@ -33,7 +36,17 @@ object Lorre extends App with LazyLogging {
...
+
+  //whatever happens we try to clean up
+  sys.addShutdownHook(shutdown())
+
+  private[this] def shutdown(): Unit = {
+    logger.info("Doing clean-up")
+    db.close()
+    Await.result(system.terminate(), 10.seconds)
+    logger.info("All things closed")
+  }
```
[the remark this answers is not in the document]
Comment (ivanopagano) [reply src/main/scala/tech/cryptonomic/conseil/Lorre.scala]: closing the system also orderly terminates his running actors, waiting for all pending messages to be handled.

It's a clean shutdown of the actor system

**Pull request merged**